### DISCUSSION TITLE
Hotspot having trouble with Android (and only Android)

### DISCUSSION THREAD

**Discussion by ark-ie** [iiab/iiab#3742] [Q&A] [answered]
Hi there! Recently set up iiab on a Raspberry Pi 4. I purchased a used Lenovo Tab M8 (2nd gen) to use exclusively for iiab. Unfortunately, I have been troubleshooting it for some time and I can’t get it to connect to the iiab hotspot properly. Here’s what I’m seeing and what I’ve tried:

The Lenovo tablet is running Android 10. It connects to my regular wifi without problems.

The Internet in a Box wifi shows up as an option in the WLAN list. But when I select it, its status rapidly switches between “connected,” “obtaining IP address,” “connected, no internet,” “saved,” and “disconnected.” It cycles through those options in 2-3 seconds. It doesn’t stay on “connected” long enough to actually show a wifi signal.

When I switch the tablet to my home wifi network I can access http://box/ and get to iiab that way. But not through the Internet in a Box hotspot.

I can connect to the hotspot from my iPhone and Mac with no issues, and access all iiab features.

One time on the tablet I accessed iiab and it showed this message, or something like it: “iiab home page menu.json file is broken and could not be patched.” Unfortunately I forgot to make a note of what I did to get that message and I can’t seem to replicate it. Since then I have accessed iiab normally through my home wifi network as above, and on the Internet in a Box wifi I have not been able to connect it long enough to pull the page up. I have accessed iiab several times through my other devices without getting that message.

I’ve run sudo iiab-hotspot-off and -on a couple of times to see if that made any difference. Rebooted both the Pi and the tablet a few times. No dice.

I’ve reset the WLAN and bluetooth settings on the tablet. Also tried a factory reset. Nada.

The Raspberry Pi is connected to ethernet, if that makes a difference.

These are some of the network settings in iiab admin. 
Server’s role in network: Gateway
Dual use internal wifi adapter: checked
Static WAN instead of DHCP: not checked
Captive portal: not checked

I pulled a diagnostics file, if that’s helpful. The link is:
/etc/iiab/diag/8.2_raspbian-12_2024-05-18_17:34:58_CDT_ark

I hope there’s a clue above, because I’m at the limit of my (very limited) knowledge. Trying to expand my tech horizons but I’m a newbie so bear with me.

Thanks! Salut

**Comment by holta** [comment]
Hi @ark-ie,

> I pulled a diagnostics file, if that’s helpful. The link is:
> /etc/iiab/diag/8.2_raspbian-12_2024-05-18_17:34:58_CDT_ark

After you run 'iiab-diagnostics' can you hit Enter, and then reply to us with the **sprunge.us** pastebin link?

**Reply by ark-ie** [reply]
Got it working based on Tim's suggestion below, but thanks for reaching out! It was throwing a 404 error whenever I tried to submit to pastebin, if you know of a solve for that let me know. 
Attaching results below in case you're interested (I took a photo of the results and copied from that, so hopefully not too many mistakes):

0. HM + Sw Quick Summary 1s -ltr /etc/iiab/install-flags # IIAB install flags
1. ﻿﻿﻿Files Specially Requested: (from "liab-diagnostics PATH/FILE1 PATH/FILEZ")
2. ﻿﻿﻿Regular Files: / .ilab-image
/etc/11ab/iiab.env
/etc/iiab/iiab.ini
/etc/iiab/local_vars.yml
/etc/iiab/11ab_state-yml
/etc/resolv.conf /etc/network/interfaces
/etc/hostapd/hostapd.conf
/etc/pa_supplicant/wa_supplicant.conf
/library/w/html/home/menu.json

3. Content of Directories: (1-level deep)

/etc/network/interfaces.d
/etc/systemd/network
/etc/NetworkManager/system-connections
/etc/netplan

4. Output of Commands:
 uname -a # Linux kernel
free # RAM memory 1scpu # CPU details
df -h # Disk usage
df -ah # Disk usage detail 1sblk # Partition mount points
blkid # Mount point details /etc/fstab lshw -C network # Network hardware/interfaces
ip addr # Network interfaces ifconfig # Network interfaces (old view) ip route # Routing table
netstat -rn # Routing table (old view)
bridge -d link # Bridge for LAN side
sudo netstat -natp # Ports/Services in use 
systemctl status dnsmasq # Is dnsmasq running?
sudo journalctl -b 0 -u dnsmasq # dnsmasq log
networkctl # systemd-networkd status
nmcl1 d # NetworkManager status
sudo fournalct1 -b 0 -u networkd-dispatcher # networkd-dispatcher log
rfkill list # Are Wifi and Bluetooth interfaces blocked?
iw reg get # Detected Wifi country code / legal frequencies
iw dev # List wireless interfaces
iw list # List capabilities of all wireless devices
systemct1 status hostapd # Downstream Wi-Fi: Is hostapd running?
1s -1 /etc/wpa_supplicant # Upstream Wi-Fi
ps -AH # Process hierarchy: staging of hostapd & wpa_supplicant?
dmesg | grep -i -e 80211 -e 802\.11 -e wireless -e wifi -e wlan -e broadcom -e brcm -e bcm -e realtek | head -100 # Wi-Fi firmware/driver msgs
1spci -nn # Devices on PCI buses
1s -1 /lib/firmware/cypress/*43430* # RPi Zero W & 3 WiFi firmware
1s -1 /lib/firmware/cypress/*43455* # RPi 3 B+ & 4 WiFi firmware
env # Environment variables
node -v # Node.js version
npm -v # npm version
/opt/i1ab/kiwix/bin/kiwix-serve --version # kiwix-tools
cd/usr/local/calibre-web-py3; sudo git log --graph - -oneline - -decorate | head -50 # Calibre-Web version
systemct1 status calibre-web # Is Calibre-Web running?
journalct1 -u calibre-web | tail -100 # Calibre-Web systemd log
/var/log/calibre-web. log
/var/log/xklb. log
journalct1 -t IIAB-CMDSRV # Admin Console CMDSRV log

5. Firewall Rules:

sudo iptables-save # Firewall rules
sudo ufw status verbose # Firewall status & rules
sudo: ufw: command not found

6. Log Files: (e.g. last 100 lines of each)
grep -B2 "'SEE ERROR ABOVE" /opt/liab/i1ab/*. log # for skip_role_on_error
/opt/ilab/i1ab/ilab-install.log
/opt/iiab/i1ab/11ab-configure.log
/opt/11ab/i1ab/i1ab-debug. log
/opt/iiab/iiab/i1ab-network.log
/opt/11ab/11ab-admin-console/admin-install.log
/var /log/messages
/var/log/syslog

**Comment by tim-moody** [comment, marked as answer]
It sounds like the tablet does not accept a hotspot with no internet access. I wonder if that is configurable on the tablet. I also wonder what would happen if you turned on captive portal.

In Admin Console check Captive Portal then click Save Configuration then click Install Configured Options.

**Reply by ark-ie** [reply]
The captive portal idea did the trick! Thanks Tim!
